### DISCUSSION TITLE
How do you train for styles now?

### DISCUSSION THREAD

**Discussion by Bonjonbon** [TheLastBen/fast-stable-diffusion#1503] [Q&A] [answered]
Hello, I've been using this colab (which works great by the way!) to train for styles but now I'm stuck. For awhile now, I been looking through previous discussions and the step by step tutorial on how to properly train a model on a style, but with the changes to learning rate, I feel like I need some clarification or correction on a few things to make things better. 

Also I'm training on a custom model if that makes any differences.

Instance: I'm using 40. should I lower it to get the best results?
Caption:  I haven't been utilizing this as I don't know if it helps with style training.
Concept: I'm using 1,330. I don't think changing this affect much, but just wanted to give you an idea of what I'm working with.

training: I did 4,000 steps with a rate of 1e-5, I think that the rate was the previous default.
Text encoder: I did 650
Concept encoder: I did a 1000
Style training: check box: I know what it says check it after training at least a 1000 times but I also seen someone say check it before you start. which one is it?

**Comment by Excalibro1** [comment, marked as answer]
1e-5 is a really fast learning rate so less training steps required to learn the concept you're training on so 4k steps might be way too high for 40 images, the text encoder is a bit hard to have a set number to train on again 650 is a lot for 40 images but this really depends on if your style is coming through with your token if its not you probably need more training to the text encoder, concept training steps seems like 1000 is a good start for your 1300 images, and I havnt used style training more than once but I'm pretty sure its right to do it after training the unet for a bit as it doesn't pass the token to the textual embedding helping to not overfit the model. Hope this helps and I am by all means no expert so don't expect it to be a perfect rundown haha

**Reply by Bonjonbon** [reply]
Thanks for the answers. Yea I'm still new to this and have been experimenting a lot with this dreambooth for styles.  I wanted to make sure I doing the it right.

**Comment by eliasprompt** [comment]
Can you share the link for that step by step tutorial you mentioned on first post? I'm interested to learn how to train styles too.

Thanks you in advice

**Reply by Bonjonbon** [reply]
Oh I'm sorry, I meant the Step by Step guide by Excalibro1 on  TheLastBen /
fast-stable-diffusion page at the top. Everything else I learned from the various topics on styles made in the discussion tab. Go there and type in style in the search bar to bring up the results. Sorry I couldn't provide anything more.
